connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/ParquetQueryVisitor.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags and
+ * the COPYRIGHT.txt file distributed with this work.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.teiid.translator.parquet;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Stack;
+
+import org.teiid.language.*;
+import org.teiid.language.visitor.HierarchyVisitor;
+import org.teiid.metadata.Column;
+import org.teiid.metadata.Table;
+import org.teiid.translator.TranslatorException;
+
+public class ParquetQueryVisitor extends HierarchyVisitor {
+
+    protected Stack<Object> onGoingExpression = new Stack<Object>();
+    private List<String> projectedColumnNames = new ArrayList<String>();
+    protected ArrayList<TranslatorException> exceptions = new ArrayList<TranslatorException>();
+
+    public HashMap<String, String> getColumnValues() {
+        return columnValues;
+    }
+
+    private HashMap<String, String> columnValues = new HashMap<>();
+
+    private Table table;
+    private String parquetPath;
+
+    public List<String> getProjectedColumnNames() {
+        return projectedColumnNames;
+    }
+
+    public ArrayList<TranslatorException> getExceptions() {
+        return exceptions;
+    }
+
+    public Table getTable() {
+        return table;
+    }
+
+    public String getParquetPath() {
+        return parquetPath;
+    }
+
+    @Override
+    public void visit(NamedTable obj) {
+        this.table = obj.getMetadataObject();
+        this.parquetPath = this.table.getProperty(ParquetMetadataProcessor.FILE);
+    }
+
+    @Override
+    public void visit(ColumnReference obj) {
+        this.onGoingExpression.add(obj.getMetadataObject());
+    }
+
+    @Override
+    public void visit(DerivedColumn obj) {
+        visitNode(obj.getExpression());
+
+        createProjectedColumn();
+    }
+
+    private void createProjectedColumn() {
+        Column column = (Column) this.onGoingExpression.pop();
+        this.projectedColumnNames.add(column.getSourceName());
+    }
+
+    @Override
+    public void visit(Comparison obj) {
+        if(this.table.getProperty(ParquetMetadataProcessor.PARTITIONING_SCHEME) == null){
```

Comment:
you can remove this consideration from here and just inline the directoryBasedPartitioning method - you are simply trying to build the column / value map, then handle that after the visitor run.

Reply:
What I understand from this is that we might use the map even for non-partitioned column comparisons and thus there is no need for that, right?

Reply:
Yes that is correct.  You can handle how the predicates are supposed to be interpreted after visitation and that will keep the visitor logic much simpler.